extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js

```diff
@@ -465,23 +491,48 @@ export class AmpLightboxViewer extends AMP.BaseElement {
     this.updateInViewport(dev().assertElement(this.container_), true);
     this.scheduleLayout(dev().assertElement(this.container_));
 
-    this.currentElementId_ = element.lightboxItemId;
-    // Hack to access private property. Better than not getting
-    // type checking to work.
-    /**@type {?}*/ (this.carousel_).implementation_.showSlideWhenReady(
-        this.currentElementId_);
-
     this.win.document.documentElement.addEventListener(
         'keydown', this.boundHandleKeyboardEvents_);
 
-    return Promise.resolve();
+    return this.resources_.requireLayout(this.carousel_)
+        .then(() => {
+          this.currentElemId_ = element.lightboxItemId;
+          /**@type {?}*/ (this.carousel_).implementation_
+              .showSlideWhenReady(this.currentElemId_);
+          this.resizeImageViewerDimensions_();
+          this.updateDescriptionBox_();
+        });
+  }
+
+  resizeImageViewerDimensions_() {
+    const imgViewer = this.elementsMetadata_[this.currentElemId_].imageViewer;
+    imgViewer.measure();
+
+    // TODO (cathyzhu): need to unregister this on slide change
+    this.unlistenViewport_ = this.getViewport().onChanged(() => {
+      // In IOS 10.3, the measured size of an element is incorrect if the
```

Comment:
let's see if 11 also needs this hack?

Reply:
- [ ] check to see if this bug repros on ios 11.